### PR TITLE
Phantom js tester

### PR DESCRIPTION
**Pull request by FelixNeutatz**
This should be sufficient for testing purposes :)

### CODE REVIEW

File: extraction/pom.xml [hunk 1]
```diff
@@ -2,147 +2,235 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <parent>
```
Comment (sscdotopen) [inline extraction/pom.xml]: you broke the 2 space indents here, please fix that (check that your Intellij is configured correctly for Java and XML)

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
+package io.ssc.trackthetrackers.extraction.resources;
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: please add the LicenseHeader here

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+  private static final Logger LOG = LoggerFactory.getLogger(GhostDriverExtractor.class);
+
+  private final URLNormalizer urlNormalizer = new URLNormalizer();
+
+
+  public static PhantomJSDriver setup(){
+    Capabilities capabilities = new DesiredCapabilities().phantomjs();
+    // Set PhantomJS Path
+    ((DesiredCapabilities) capabilities).setCapability("phantomjs.binary.path", "/home/felix/Software/phantomjs/bin/phantomjs");
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: use a config parameter here please

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    ((DesiredCapabilities) capabilities).setCapability("phantomjs.settings.loadImages", false);
+    //((DesiredCapabilities) capabilities).setCapability("phantomjs.settings.localToRemoteUrlAccessEnabled", false);
+
+    return new PhantomJSDriver(capabilities);
+  }
+
+  public synchronized Iterable<Resource> extractResources(String sourceUrl, String html) {
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: no need to have synchronized here

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    return extractResources(sourceUrl, html, null);
+  }
+
+  public synchronized Iterable<Resource> extractResources(String sourceUrl, String html, PhantomJSDriver phantom) {
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: no need to have synchronized here

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+        if(tag.data().length() > 1 && tag.data().contains("function")) {
+          scriptHtml.add(tag.data());
+        }
+      }*/
+    }
+
+    scriptHtml.add(html);
+
+
+    for(String shtml : scriptHtml) {
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: no space after for

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+      File temp = null;
+      try {
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: the try block should ensure that the file is always closed

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+        bw.close();
+
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+
+
+      File tempLog = null;
+      do {
+        try {
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: the try block should ensure that the file is always closed

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+          //tempLog = File.createTempFile("log", ".log");
+          tempLog = new File("temp.log");
+          tempLog.createNewFile();
+        } catch (Exception e) {
+          e.printStackTrace();
+        }
+      }while(!tempLog.exists());
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: space before while

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+
+      if(phantom == null) {
```
Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: space after if

File: extraction/src/main/java/org/openqa/selenium/browserlaunchers/Proxies.java [hunk 11]
```diff
@@ -0,0 +1,10 @@
+package org.openqa.selenium.browserlaunchers;
```
Comment (sscdotopen) [inline extraction/src/main/java/org/openqa/selenium/browserlaunchers/Proxies.java]: please add a license header here

File: extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java (hunk 7 above)
Comment (FelixNeutatz) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: I don't understand this point? java.io.File has no close() method?!

Comment (sscdotopen) [inline extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java]: sry, my comment was unclear. The scanner that uses the file (and the delete() call on the file) should be done in a finally block

**Comment by FelixNeutatz** [comment]
done, the great thing is that you don't need to install any additional libraries. Just run mvn :)

**Pull request merged**